replace_with/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+use std::ptr;
+
+pub fn replace_with<T, F>(t: &mut T, f: F) where F: FnOnce(T) -> T
+{
+	unsafe
+	{
+		let t_ptr = t as *mut T;
+		ptr::write(t, f(ptr::read(t_ptr)));
```

Comment:
I committed again to change t to t_ptr. I have no idea if this would actually work as-is or not, but RUST wasn't giving me a type error sooooooo....